src/RunTests/RunVSTestTask.cs

```diff
@@ -0,0 +1,376 @@
+        /// Gets or Sets source blame on test failure.
+        /// </summary>
+        public string VSTestBlame { get; set; }
+
+        /// <summary>
+        /// Gets or Sets source blame on test crash.
+        /// </summary>
+        public string VSTestBlameCrash { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Dumptype used for crash source blame.
+        /// </summary>
+        public string VSTestBlameCrashDumpType { get; set; }
+
+        /// <summary>
+        /// Gets or Sets source blame on test crash even if test pass.
+        /// </summary>
+        public string VSTestBlameCrashCollectAlways { get; set; }
+
+        /// <summary>
+        /// Gets or Sets source blame on test hang.
+        /// </summary>
+        public string VSTestBlameHang { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Dumptype used for hang source blame.
+        /// </summary>
+        public string VSTestBlameHangDumpType { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Time out for hang source blame.
+        /// </summary>
+        public string VSTestBlameHangTimeout { get; set; }
+
+        /// <summary>
+        /// Gets or Sets The directory path where trace data collector is.
+        /// </summary>
+        public string VSTestTraceDataCollectorDirectoryPath { get; set; }
+
+        /// <summary>
+        /// Gets or Sets disabling Microsoft logo while running test through VSTest.
+        /// </summary>
+        public string VSTestNoLogo { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Test artifacts processing mode which is applicable for .NET 5.0 or later versions.
+        /// </summary>
+        public string VSTestArtifactsProcessingMode { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Correlation Id of test session.
+        /// </summary>
+        public string VSTestSessionCorrelationId { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Runner version of VSTest.
+        /// </summary>
+        [Required]
+        public string VSTestRunnerVersion { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Path to nuget package cache.
+        /// </summary>
+        [Required]
+        public string NugetPath { get; set; }
+
+        protected override string ToolName => "vstest.console.exe";
+
+        protected override MessageImportance StandardOutputLoggingImportance => MessageImportance.High;
+
+        protected override MessageImportance StandardErrorLoggingImportance => MessageImportance.High;
+
+        /// <inheritdoc/>
+        protected override string GenerateFullPathToTool()
+        {
+#if NET6_0_OR_GREATER
+            return $@"{NugetPath}microsoft.testplatform\{VSTestRunnerVersion}\tools\net6.0\Common7\IDE\Extensions\TestPlatform\vstest.console.exe";
+#else
+            return $@"{NugetPath}microsoft.testplatform\{VSTestRunnerVersion}\tools\net462\Common7\IDE\Extensions\TestPlatform\vstest.console.exe";
```

Comment:
Seems to me like we should just mimic what `dotnet test` does for the `dotnet` scenario, and for the VS scenario just find vstest from the VS install. Digging through the package cache seems like a pretty brittle thing to do, especially since the versions will need to align with some PackageReference for it to work consistently.

Reply:
In fact, for the `dotnet` case can we actually just invoke the target which `dotnet test` does?

Reply:
In `C:\Program Files\dotnet\sdk\...\Microsoft.TestPlatform.targets` I see:

```
<Target Name="VSTest" DependsOnTargets="ShowInfoMessageIfProjectHasNoIsTestProjectProperty">
```

So we should probably just have something like this:
```xml
<Target Name="RunVSTestAfterTest" AfterTargets="Test" DependsOnTargets="VSTest" />
```

And then in the non-`dotnet` case, figure out how to import `Microsoft.TestPlatform.targets` or something. This would probably only work for SDK-style projects, but that's probably OK?

That way we never are maintaining how VSTest is invoked at all and it's always consistent with `dotnet test`.

Reply:
> And then in the non-dotnet case, figure out how to import Microsoft.TestPlatform.targets or something. 
Microsoft.TestPlatform.targets invokes a task that invokes dotnet